starter-code/tests/FunctionsAndArraysSpec.js

```diff
@@ -1,246 +1,418 @@
+//     ];
+//     expect(greatestProduct(matrix)).toBe(1);
+//   });
+
+//   it('Return 16 when all the numbers of the arrays are 2', function () {
+//     var matrix = [
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2]
+//     ];
+//     expect(greatestProduct(matrix)).toBe(16);
+//   });
+// });
+
+
+
+
+
+
+
+
+
+
+// ---------  My Answers Below ------------------------------------------------------------------------------
+
+// David Mudd & Ali Sattar
+
+// function maxOfTwoNumbers(num1, num2) {
+//   if (num1 > num2) {
+//     console.log(`${num1} is larger than ${num2}.`);
+//   } else {
+//     console.log(`${num2} is larger than ${num1}.`)
+//   }
+// }
+
+// maxOfTwoNumbers(12,11);
+
+// --------------------------
+
+// var words = [
+//   'mystery',
+//   'brother',
+//   'aviator',
+//   'crocodile',
+//   'pearl',
+//   'orchard',
+//   'crackpot'
+// ];
+
+
+// var longest = "";
+
+// words.forEach(function(oneWord) {
+// if(longest.length < oneWord.length){
+//   longest = oneWord
+// }
+// return longest
+// });
+// console.log("longest is: ", longest);
+
+// ------------------------------
+
+// var numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
+
+
+// function calculateSum(arrayOfNumbers) {
+
+//   var sum = 0;
+
+//   arrayOfNumbers.forEach(function(oneNumber) {
+//    sum += oneNumber;
+//   });
+
+//   return sum;
+// };
+
+// console.log(calculateSum(numbers));
+
+
+// ------------------------
 
+// var numbers = [2, 6, 9, 10, 7, 4, 1, 9];
+
+// // function calcAvg(numberArray) {
+// //   var sum = 0;
+// // }
+
+
+// var sum = 0;
```

Comment:
if you put the sum variable inside of the sumBlah function then you can have sum reset to 0 each time you call the function. Otherwise having sum as a global variable it never really resets unless you make `sum = 0` inside of of your function.